persist/fs/commitlog/bitset.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+const (
+	wordSize    = uint(64)
+	logWordSize = uint(6) // lg(wordSize)
+)
+
+type bitSet struct {
+	values []uint64
+}
+
+func newBitSet(size uint) *bitSet {
+	return &bitSet{values: make([]uint64, bitSetIndexOf(size)+1)}
+}
+
+func (b *bitSet) test(i uint) bool {
+	idx := bitSetIndexOf(i)
+	if idx >= len(b.values) {
+		return false
+	}
+	return b.values[idx]&(1<<(i&(wordSize-1))) != 0
```

Comment:
Shouldn't this just be `b.values[idx] & (i & (wordSize - 1))`? e.g., if i is 127, you want to bit and with 63, not 1<<63, no?

Also would suggest making `wordSize - 1` a const and name it `wordMask`, I don't trust compilers but your call.